server/routes/createGoal/createGoalController.ts

```diff
@@ -141,24 +141,37 @@ export default class CreateGoalController {
   }
 
   getAddNoteView: RequestHandler = async (req, res, next): Promise<void> => {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { goalIndex } = req.params
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { mode } = req.query
     const { prisonerSummary } = req.session
-    const addNoteForm = req.session.newGoal.addNoteForm || {}
+
+    if (isEditMode(req)) {
+      req.session.newGoal = {
+        addNoteForm: req.session.newGoals[parseInt(goalIndex, 10) - 1].addNoteForm,
+      } as NewGoal
+    } else if (!req.session.newGoal.addNoteForm) {
+      req.session.newGoal.addNoteForm = {}
+    }
+
+    const { addNoteForm } = req.session.newGoal
 
     const view = new AddNoteView(prisonerSummary, addNoteForm, req.flash('errors'))
-    res.render('pages/goal/add-note/index', { ...view.renderArgs })
+    return res.render('pages/goal/add-note/index', { ...view.renderArgs })
```

Comment:
I know the nunjucks view for the Add Note page doesnt need it (yet? 🙄 ), but I wonder if it would be worth adding the `isEditMode` flag to the class `AddNoteView` in the same way you have done with the other view classes here?

Like I said, I know we don't need it .... but it wouldn't surprise me if we did need it at some point (you know how design continually change their mind! 🙄 ), and also it would be consistent with the other view classes here.

What do you think?

Reply:
Makes sense for consistency and as you said no doubt we might need it in future!